snippets/optional-parameters-in-the-middle.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Optional parameters in the middle
+author: irustm
+level: beginner
+tags:
+  - routing
+---
+# Content
+Navigate with matrix params:
+
+the router will navigate to `/first;name=foo/details`
```

Comment:
What would be the differences/benefits compared to `?name=foo`?

Reply:
@kirjs as far as I know Angular takes query params as different urls, so when you make link active with `routerLinkActive`  it'll be not highlighted for all cases.

Optional (matrix) params don't have this problem

Reply:
You will get matrix params only in the necessary component. And if there is query params, you will get them in all components. Example: https://stackblitz.com/edit/angular-xvy5pd